### ISSUE TITLE
Datagrid default filter's input field does not have associated label

### ISSUE THREAD

**Issue by aognianova** [vmware/clarity#4554]
## Describe the bug

This is an issue with accessibility. We got a report from AccessLevel where they note that there is no label associated to the input field in the filter popover. When form fields do not have a programmatic (accessible) label, assistive technologies may incorrectly render the label or provide no label at all to users. When labels are not present or are incorrect, users of assistive technologies may not be able to complete a form.
Their recommendation is that developers can define form labels using the label element whenever an on-screen label is present. The label element's value is then explicitly associated with the relevant form field using the for and id attributes. When on-screen text is not available or can not programmatically be contained in a label, developers can use another method to provide an accessible name such as by using the title, aria-label, or aria-labelledby attribute to provide the label.

## How to reproduce

Simply use clr-dg-column component with default filter.

1. Go to a view containing clr-dg-column (e.g. https://clarity.design/documentation/datagrid/built-in-filters)
2. Click on filter icon
3. Right-Click on the filter field inside the popup and choose Inspect.
4. Check its accessibility properties in the debugger:
<input type="text" name="search" class="clr-input ng-pristine ng-valid ng-touched" ng-reflect-name="search" ng-reflect-model="">

## Expected behavior

Input field inside the popup should have associated label or aria-label.

## Versions

**App**

- Angular: [9]
- Node: [10.16.1]
- Clarity: [3.0.0]

**Device:**

- Type: [PC]
- OS: [Win10]
- Browser [chrome]
- Version [81]

**Comment by chlane** [comment]
This has been reported with accessibility input by @gkhadK in https://github.com/vmware/clarity/issues/4664

**Comment by gnomeontherun** [comment]
Closing as duplicate.
